### PR TITLE
vscode-extensions.pkief.material-{icon-theme,product-icons}: bump versions

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

- vscode-extensions.pkief.material-product-icons: 1.5.0 -> 1.7.0
- vscode-extensions.pkief.material-icon-theme: 4.31.0 -> 4.34.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by superherointj** [comment]
@eclairevoyant Please, update the package to latest and complete the meta fields.

**Comment by superherointj** [comment]
It should be a single commit per package. Please, squash the commits.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -3327,23 +3327,29 @@ let
         mktplcRef = {
           name = "material-icon-theme";
           publisher = "PKief";
-          version = "4.31.0";
-          sha256 = "0rn4dyqr46wbgi4k27ni6a6i3pa83gyaprhds5rlndjaw90iakb4";
+          version = "4.34.0";
+          sha256 = "sha256-xxOEUvMjqJbl8lONB/So2NoIAVPOxysTq2YQY3iHGqo=";
         };
         meta = {
+          description = "Material Design file/folder icon theme for VSCode";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:3334]: `changelog` is missing

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -3327,23 +3327,29 @@ let
...
         mktplcRef = {
           name = "material-product-icons";
           publisher = "PKief";
-          version = "1.5.0";
-          sha256 = "sha256-gKU21OS2ZFyzCQVQ1fa3qlahLBAcJaHDEcz7xof3P4A=";
+          version = "1.7.0";
+          sha256 = "sha256-F6sukBQ61pHoKTxx88aa8QMLDOm9ozPF9nonIH34C7Q=";
         };
         meta = {
+          description = "Material Design product icon theme for VSCode";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:3349]: `changelog` is missing

**Comment by superherointj** [comment]
@eclairevoyant Can you rebase this?

**Comment by superherointj** [comment]
https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vscode/extensions/README.md
Would you mind following these guidelines for this PR?

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 3]
```diff
@@ -3327,23 +3327,29 @@ let
         mktplcRef = {
           name = "material-icon-theme";
           publisher = "PKief";
-          version = "4.31.0";
-          sha256 = "0rn4dyqr46wbgi4k27ni6a6i3pa83gyaprhds5rlndjaw90iakb4";
+          version = "4.34.0";
+          sha256 = "sha256-xxOEUvMjqJbl8lONB/So2NoIAVPOxysTq2YQY3iHGqo=";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:3331]:
```suggestion
          hash = "sha256-xxOEUvMjqJbl8lONB/So2NoIAVPOxysTq2YQY3iHGqo=";
```

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 4]
```diff
@@ -3327,23 +3327,29 @@ let
...
       };
 
       pkief.material-product-icons = buildVscodeMarketplaceExtension {
         mktplcRef = {
           name = "material-product-icons";
           publisher = "PKief";
-          version = "1.5.0";
-          sha256 = "sha256-gKU21OS2ZFyzCQVQ1fa3qlahLBAcJaHDEcz7xof3P4A=";
+          version = "1.7.0";
+          sha256 = "sha256-F6sukBQ61pHoKTxx88aa8QMLDOm9ozPF9nonIH34C7Q=";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:3346]:
```suggestion
          hash = "sha256-F6sukBQ61pHoKTxx88aa8QMLDOm9ozPF9nonIH34C7Q=";
```

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 5]
```diff
@@ -3327,23 +3327,29 @@ let
...
         };
         meta = {
+          description = "Material Design product icon theme for VSCode";
+          homepage = "https://github.com/PKief/vscode-material-product-icons/blob/main/README.md";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:3350]:
```suggestion
          homepage = "https://github.com/PKief/vscode-material-product-icons/";
```

**Comment by eclairevoyant** [comment]
> It should be a single commit per package. Please, squash the commits.

No, they're separate logical changes.
Anyway, best to close this PR until we have more sensible defaulting for vscode extensions, I guess.

**Pull request closed without merging**